src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -996,5 +996,12 @@ internal static unsafe uint SysStringByteLen(IntPtr s)
 
         [SupportedOSPlatform("windows")]
         public static Type? GetTypeFromCLSID(Guid clsid) => GetTypeFromCLSID(clsid, null, throwOnError: false);
+
+        /// <summary>
+        /// Initializes the underlying handle of a newly created <see cref="SafeHandle" /> to the provided value.
+        /// </summary>
+        /// <param name="safeHandle"><see cref="SafeHandle"/> instance to update</param>
+        /// <param name="handle">Pre-existing handle</param>
+        public static void InitHandle(SafeHandle safeHandle, IntPtr handle) => safeHandle.SetHandle(handle);
```

Comment:
Presumably in the name of efficiency for P/Invoke marshaling we're ok with this null ref'ing rather than throwing an ArgumentNullException if safeHandle is null?

Reply:
Yes.

Reply:
Might be worth a short comment to that effect.